Sources/Brave/WebFilters/FilterListResourceDownloader.swift

```diff
@@ -40,23 +40,27 @@ public actor FilterListResourceDownloader {
   /// - Warning: This method loads filter list settings.
   /// You need to wait for `DataController.shared.initializeOnce()` to be called first before invoking this method
   public func loadFilterListSettingsAndCachedData() async {
-    guard let folderURL = await FilterListSetting.makeFolderURL(
-      forFilterListFolderPath: Preferences.AppState.lastDefaultFilterListFolderPath.value
-    ), FileManager.default.fileExists(atPath: folderURL.path) else {
-      return
+    if let defaultFilterListFolderURL = await FilterListSetting.makeFolderURL(
+      forFilterListFolderPath: Preferences.AppState.lastFilterListCatalogueComponentFolderPath.value
+    ), FileManager.default.fileExists(atPath: defaultFilterListFolderURL.path),
+       let resourcesFolderURL = await FilterListSetting.makeFolderURL(
+         forFilterListFolderPath: Preferences.AppState.lastAdBlockResourcesFolderPath.value
+       ), FileManager.default.fileExists(atPath: resourcesFolderURL.path) {
+      let resourcesInfo = await didUpdateResourcesComponent(folderURL: resourcesFolderURL)
+      async let startedCustomFilterListsDownloader: Void = FilterListCustomURLDownloader.shared.startIfNeeded()
+      async let cachedFilterLists: Void = compileCachedFilterLists(resourcesInfo: resourcesInfo)
+      async let compileDefaultEngine: Void = compileDefaultEngine(defaultFilterListFolderURL: defaultFilterListFolderURL, resourcesInfo: resourcesInfo)
+      _ = await (startedCustomFilterListsDownloader, cachedFilterLists, compileDefaultEngine)
+    } else if let legacyComponentFolderURL = await FilterListSetting.makeFolderURL(
+      forFilterListFolderPath: Preferences.AppState.lastLegacyDefaultFilterListFolderPath.value
+    ), FileManager.default.fileExists(atPath: legacyComponentFolderURL.path) {
+      // TODO: @JS Remove this after this release. Its here just so users can upgrade without a pause to their adblocking
```

Comment:
what does it mean `after this release`?

Reply:
oh its just temporary code that we can remove next version.